MRETestBed/Assets/TestBed Assets/Scripts/VertexMaterialPatcher.cs

```diff
@@ -1,16 +1,69 @@
-// Copyright (c) Microsoft Corporation. All rights reserved.
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 using MWAssets = MixedRealityExtension.Assets;
+using MixedRealityExtension.Patching.Types;
 
+using MixedRealityExtension.API;
+using System;
+using System.Collections.Generic;
 using UnityEngine;
 using UnityEngine.Rendering;
 
 public class VertexMaterialPatcher : MixedRealityExtension.Factories.DefaultMaterialPatcher
 {
+	private static int EmissiveColorProp = Shader.PropertyToID("_EmissiveColor");
+	private static int EmissiveTexProp = Shader.PropertyToID("_EmissiveTex");
+
+	protected Dictionary<int, Guid> emissiveTextureAssignments = new Dictionary<int, Guid>(20);
+
 	public override void ApplyMaterialPatch(Material material, MWAssets.Material patch)
 	{
 		base.ApplyMaterialPatch(material, patch);
 
+		if (patch.EmissiveColor != null)
+		{
+			var color = material.GetColor(EmissiveColorProp);
+			color.r = patch.EmissiveColor.R ?? color.r;
+			color.g = patch.EmissiveColor.G ?? color.g;
+			color.b = patch.EmissiveColor.B ?? color.b;
+			color.a = patch.EmissiveColor.A ?? color.a;
+			material.SetColor(EmissiveColorProp, color);
```

Comment:
Note: the patch utils are internal to the DLL, so they can't be used in factories.